src/components/profile/ViewAvatar.js

```diff
@@ -12,23 +15,44 @@ const ViewAvatar = props => {
   const { styles } = props
   const store = GDStore.useStore()
   const profile = store.get('profile')
+  const wrappedUserStorage = useWrappedUserStorage()
+  const [showErrorDialog] = useErrorDialog()
 
   const handleCameraPress = event => {
-    event.stopPropagation()
+    event.preventDefault()
     props.screenProps.push('EditAvatar')
   }
 
   const handleClosePress = event => {
-    event.stopPropagation()
-    props.screenProps.push('Profile')
+    event.preventDefault()
+    wrappedUserStorage.setProfileField('avatar', null, 'public').catch(e => showErrorDialog('Saving image failed', e))
+  }
+
+  const handleAddAvatar = avatar => {
+    wrappedUserStorage.setProfileField('avatar', avatar, 'public').catch(e => showErrorDialog('Saving image failed', e))
+    props.screenProps.push('EditAvatar')
   }
 
   return (
     <Wrapper>
       <Section style={styles.section}>
-        <UserAvatar profile={profile} size={272} />
-        <CircleButtonWrapper style={styles.closeButton} iconName={'close'} iconSize={20} onPress={handleClosePress} />
-        <CameraButton style={styles.cameraButton} handleCameraPress={handleCameraPress} />
+        {profile.avatar ? (
```

Comment:
Is it not better to reduce some ifs, and do it like this:

```
{profile.avatar ? (
    <>
        <UserAvatar profile={profile} size={272} />
        <CircleButtonWrapper style={styles.closeButton} iconName={'close'} iconSize={20} onPress={handleClosePress} />
        <CameraButton style={styles.cameraButton} handleCameraPress={handleCameraPress} />
    </>
) : (
    <>
        <InputFile onChange={handleAddAvatar}>
           <UserAvatar profile={profile} size={272} />{' '}
        </InputFile>
        <InputFile onChange={handleAddAvatar}>
            <CameraButton style={styles.cameraButton} />
        </InputFile>
    </>
)}
```

Reply:
done